mm2src/mm2_bin_lib/src/mm2_native_lib.rs

```diff
@@ -109,7 +148,17 @@ pub unsafe extern "C" fn mm2_main(conf: *const c_char, log_cb: extern "C" fn(lin
         );
     }
 
-    StartupResultCode::Ok as i8
+    match init_receiver.recv_timeout(Duration::from_secs(startup_timeout)) {
+        Ok(status) => status as i8,
```

Comment:
i would argue we can have this be just `.recv()` with no timeout:
- we always reply in every error/success/panic case to `init_receiver`.
- if the thread never spawned (so will never reply) we exit with spawn error.
- if mm2 process is somehow slow starting up, well, we gotta wait for it anyway.
    - if it was slow that the timeout passed and we replied to the caller with `TimeoutError` but the mm2 thread is running, isn't this a problem?

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2389/commits/299592b2b94e1a341fb9f92f97159d3f3bd8deb0 should fix it